embulk-input-s3/src/main/java/org/embulk/input/s3/AbstractS3FileInputPlugin.java

```diff
@@ -67,6 +67,11 @@
         @ConfigDefault("null")
         public Optional<String> getAccessKeyId();
 
+        @Config("http_proxy")
+        @ConfigDefault("null")
+        public Optional<HttpProxy> getHttpProxy();
+        public void setHttpProxy(Optional<HttpProxy> httpProxy);
```

Comment:
I understand your thoughts. I think that it's good to have both of your env option and default env. The default `env` is helped for the following use cases.

- An Embulk user might have multiple Embulk config files for ingesting data from S3 to somewhere. Because there're various bucket names, paths and file formats. It doesn't make sense to define same http proxy configuration for each Embulk config file.
- The http proxy configuration is sometimes required by both input plugin and output plugin (executor plugin as well). For example, if an user ingests data from S3 to S3, he needs to declare http proxy configuration in both of `in:` and `out:`. It's better to separate the configuration from others.